rust/validation/src/validation/store.rs

```diff
@@ -88,68 +88,93 @@ impl StoreValidate<Schema> for Store {
     }
 }
 
+trait SchemaStringHelper {
+    fn process_schema_string<F>(
+        &self,
+        schema_name: &str,
+        func: F,
+    ) -> Result<ValidationResult, StoreValidateError>
+    where
+        F: FnOnce(Schema) -> Result<ValidationResult, StoreValidateError>;
+    fn process_schema_string_unfaillible<F>(&self, schema_name: &str, func: F) -> ValidationResult
+    where
+        F: FnOnce(Schema) -> ValidationResult;
+}
+
+impl SchemaStringHelper for Store {
+    fn process_schema_string<F>(
+        &self,
+        schema_name: &str,
+        func: F,
+    ) -> Result<ValidationResult, StoreValidateError>
+    where
+        F: FnOnce(Schema) -> Result<ValidationResult, StoreValidateError>,
+    {
+        match Schema::try_from(schema_name) {
+            Ok(schema_type) => func(schema_type),
+            Err(_) => Ok({
+                let mut ctx = Context::new(self, None);
+                ctx.add_violation(Violation::InvalidSchema {
+                    schema: schema_name.into(),
+                });
+                ctx.into()
+            }),
+        }
+    }
+    fn process_schema_string_unfaillible<F>(&self, schema_name: &str, func: F) -> ValidationResult
```

Comment:
Needing to have two of these makes it clear that this pattern is broken. Let's try to find a better way. I am ok with removing the schema error from the violations and instead failing the result. This means we can change the return type on the validate_value to also be a result. And then this can be dropped and we just use try_from(string)?